src/core/thread/mle_router.hpp

```diff
@@ -62,6 +64,141 @@ class NetworkDataLeader;
+    };
+
+    /**
+     * This method reads delayed response header from the message.
+     *
+     * @param[in]  aMessage  A reference to the message.
+     *
+     * @returns The number of bytes read.
+     *
+     */
+    uint16_t ReadFrom(Message &aMessage) {
+        return aMessage.Read(aMessage.GetLength() - sizeof(*this), sizeof(*this), this);
+    };
+
+    /**
+     * This method updates delayed response header in the message.
+     *
+     * @param[in]  aMessage  A reference to the message.
+     *
+     * @returns The number of bytes written.
+     *
+     */
+    int UpdateIn(Message &aMessage) {
+        return aMessage.Write(aMessage.GetLength() - sizeof(*this), sizeof(*this), this);
+    };
+
+    /**
+     * This method removes delayed response header from the message.
+     *
+     * @param[in]  aMessage  A reference to the message.
+     *
+     * @retval kThreadError_None  Successfully removed the header.
+     *
+     */
+    static ThreadError RemoveFrom(Message &aMessage) {
+        return aMessage.SetLength(aMessage.GetLength() - sizeof(DelayedResponseHeader));
+    };
+
+    /**
+     * This method returns a pointer to the next delayed message on the list.
+     *
+     * @returns  A pointer to the next delayed message.
+     *
+     */
+    Message *GetNext(void) { return mNext; };
+
+    /**
+     * This method returns a time when the message shall be sent.
+     *
+     * @returns  A time when the message shall be sent.
+     *
+     */
+    uint32_t GetSendTime(void) { return mSendTime; };
+
+    /**
+     * This method returns a destination of the delayed message.
+     *
+     * @returns  A destination of the delayed message.
+     *
+     */
+    Ip6::Address &GetDestination(void) { return mDestination; };
+
+    /**
+     * This method sets the pointer to the next delayed message.
+     *
+     * @param[in]  aMessage  A pointer to the next message.
+     *
+     */
+    void SetNext(Message *aMessage) { mNext = aMessage; };
+
+    /**
+     * This method checks if the message shall be sent before the given time.
+     *
+     * @param[in]  aTime  A time to compare.
+     *
+     * @retval TRUE   If the message shall be sent before the given time.
+     * @retval FALSE  Otherwise.
+     */
+    bool IsEarlier(uint32_t aTime) { return (static_cast<int32_t>(aTime - mSendTime) > 0); };
+
+    /**
+     * This method checks if the message shall be sent after the given time.
+     *
+     * @param[in]  aTime  A time to compare.
+     *
+     * @retval TRUE   If the message shall be sent after the given time.
+     * @retval FALSE  Otherwise.
+     */
+    bool IsLater(uint32_t aTime) { return (static_cast<int32_t>(aTime - mSendTime) < 0); };
+
+private:
+    Ip6::Address mDestination;  ///< IPv6 address of the message destination.
+    uint32_t mSendTime;         ///< Time when the message shall be sent.
+    Message *mNext;             ///< Pointer to the next delayed message. NULL if there is no more messages.
```

Comment:
The Message object has its own list pointers that you could use.  See the `MessageQueue` class for maintaining lists of messages.